# --- T2-COPYRIGHT-BEGIN ---
# t2/package/*/geany/hotfix-gcc15.patch
# Copyright (C) 2025 The T2 SDE Project
# SPDX-License-Identifier: GPL-2.0 or patched project license
# --- T2-COPYRIGHT-END ---

--- geany-2.0/scintilla/src/Indicator.cxx.vanilla	2025-04-30 19:10:39.231213740 +0200
+++ geany-2.0/scintilla/src/Indicator.cxx	2025-04-30 19:10:55.311695529 +0200
@@ -5,6 +5,7 @@
 // Copyright 1998-2001 by Neil Hodgson <neilh@scintilla.org>
 // The License.txt file describes the conditions under which this software may be distributed.
 
+#include <cstdint>
 #include <cmath>
 
 #include <stdexcept>
--- geany-2.0/scintilla/src/ViewStyle.cxx.vanilla	2025-04-30 19:11:52.781376557 +0200
+++ geany-2.0/scintilla/src/ViewStyle.cxx	2025-04-30 19:11:59.372853352 +0200
@@ -6,6 +6,7 @@
 // The License.txt file describes the conditions under which this software may be distributed.
 
 #include <cstddef>
+#include <cstdint>
 #include <cassert>
 #include <cstring>
 #include <cmath>
--- geany-2.0/scintilla/src/AutoComplete.cxx.vanilla	2025-04-30 19:09:17.052756521 +0200
+++ geany-2.0/scintilla/src/AutoComplete.cxx	2025-04-30 19:09:26.842558514 +0200
@@ -7,6 +7,7 @@
 
 #include <cstddef>
 #include <cstdlib>
+#include <cstdint>
 #include <cassert>
 #include <cstring>
 #include <cstdio>
--- geany-2.0/scintilla/src/Decoration.cxx.vanilla	2025-04-30 19:10:13.502509754 +0200
+++ geany-2.0/scintilla/src/Decoration.cxx	2025-04-30 19:10:18.540315317 +0200
@@ -6,6 +6,7 @@
 
 #include <cstddef>
 #include <cstdlib>
+#include <cstdint>
 #include <cstring>
 #include <cstdio>
 #include <cstdarg>
--- geany-2.0/scintilla/src/CellBuffer.cxx.vanilla	2025-04-30 19:09:51.679714516 +0200
+++ geany-2.0/scintilla/src/CellBuffer.cxx	2025-04-30 19:09:55.106844269 +0200
@@ -7,6 +7,7 @@
 
 #include <cstddef>
 #include <cstdlib>
+#include <cstdint>
 #include <cassert>
 #include <cstring>
 #include <cstdio>
--- geany-2.0/scintilla/src/ChangeHistory.cxx.vanilla	2025-04-30 19:10:01.840744031 +0200
+++ geany-2.0/scintilla/src/ChangeHistory.cxx	2025-04-30 19:10:07.387198080 +0200
@@ -7,6 +7,7 @@
 
 #include <cstddef>
 #include <cstdlib>
+#include <cstdint>
 #include <cassert>
 
 #include <stdexcept>
--- geany-2.0/scintilla/src/LineMarker.cxx.vanilla	2025-04-30 19:11:12.001344751 +0200
+++ geany-2.0/scintilla/src/LineMarker.cxx	2025-04-30 19:11:18.083583119 +0200
@@ -5,6 +5,7 @@
 // Copyright 1998-2011 by Neil Hodgson <neilh@scintilla.org>
 // The License.txt file describes the conditions under which this software may be distributed.
 
+#include <cstdint>
 #include <cstring>
 #include <cmath>
 
--- geany-2.0/scintilla/src/KeyMap.cxx.vanilla	2025-04-30 19:11:01.677433470 +0200
+++ geany-2.0/scintilla/src/KeyMap.cxx	2025-04-30 19:11:06.729134459 +0200
@@ -6,6 +6,7 @@
 // The License.txt file describes the conditions under which this software may be distributed.
 
 #include <cstdlib>
+#include <cstdint>
 
 #include <stdexcept>
 #include <string_view>
--- geany-2.0/scintilla/src/CallTip.cxx.vanilla	2025-04-30 19:09:35.663254896 +0200
+++ geany-2.0/scintilla/src/CallTip.cxx	2025-04-30 19:09:45.209247306 +0200
@@ -7,6 +7,7 @@
 
 #include <cstddef>
 #include <cstdlib>
+#include <cstdint>
 #include <cassert>
 #include <cstring>
 #include <cstdio>
--- geany-2.0/scintilla/src/Document.cxx.vanilla	2025-04-30 19:10:23.378371808 +0200
+++ geany-2.0/scintilla/src/Document.cxx	2025-04-30 19:10:31.824000000 +0200
@@ -7,6 +7,7 @@
 
 #include <cstddef>
 #include <cstdlib>
+#include <cstdint>
 #include <cassert>
 #include <cstring>
 #include <cstdio>
--- geany-2.0/scintilla/src/Style.cxx.vanilla	2025-04-30 19:11:39.341813236 +0200
+++ geany-2.0/scintilla/src/Style.cxx	2025-04-30 19:11:46.093772586 +0200
@@ -5,6 +5,8 @@
 // Copyright 1998-2001 by Neil Hodgson <neilh@scintilla.org>
 // The License.txt file describes the conditions under which this software may be distributed.
 
+#include <cstdint>
+
 #include <stdexcept>
 #include <string_view>
 #include <vector>
--- geany-2.0/scintilla/src/PerLine.cxx.vanilla	2025-04-30 19:11:24.398516837 +0200
+++ geany-2.0/scintilla/src/PerLine.cxx	2025-04-30 19:11:32.805861796 +0200
@@ -6,6 +6,7 @@
 // The License.txt file describes the conditions under which this software may be distributed.
 
 #include <cstddef>
+#include <cstdint>
 #include <cassert>
 #include <cstring>
 
--- geany-2.0/scintilla/src/XPM.cxx.vanilla	2025-04-30 19:12:07.821384251 +0200
+++ geany-2.0/scintilla/src/XPM.cxx	2025-04-30 19:12:14.056160391 +0200
@@ -6,6 +6,7 @@
 // The License.txt file describes the conditions under which this software may be distributed.
 
 #include <cstdlib>
+#include <cstdint>
 #include <cstring>
 #include <climits>
 
--- geany-2.0/scintilla/gtk/PlatGTK.cxx.vanilla	2025-04-30 19:08:46.348551252 +0200
+++ geany-2.0/scintilla/gtk/PlatGTK.cxx	2025-04-30 19:09:01.805109890 +0200
@@ -5,6 +5,7 @@
 
 #include <cstddef>
 #include <cstdlib>
+#include <cstdint>
 #include <cstring>
 #include <cstdio>
 #include <cmath>
